src/commands/version.js

```diff
@@ -0,0 +1,275 @@
+   * Function to increment the version in a css file
+   * @param {string} dir - File path
+   * @param {string} type - Increment type.
+   */
+  function incrementStylesCssVersion(dir, type) {
+    if (fs.existsSync(dir)) {
+      try {
+        let content = fs.readFileSync(dir, 'utf-8');
+        const versionRegex = /Version:\s*([\d]+\.[\d]+\.[\d]+(?:-[a-z]+\.[\d]+)?)/i;
+        const match = content.match(versionRegex);
+        if (match) {
+          const oldVersion = match[1];
+          const newVersion = incrementVersionString(oldVersion, type);
+          content = content.replace(versionRegex, `Version: ${newVersion}`);
+          fs.writeFileSync(dir, content, 'utf-8');
+          terminal(`#${dir}: ${oldVersion} -> ${newVersion}\n`);
+        }
+      } catch (err) {
+        console.error(`Error incrementing version in ${dir}: ${err.message}`);
+      }
+    }
+  }
+
+  /**
+   * Function to increment the version in a php file
+   * @param {string} dir - File path
+   * @param {string} type - Increment type.
+   */
+  function incrementPluginPhpVersion(dir, type) {
+    if (fs.existsSync(dir)) {
+      try {
+        let content = fs.readFileSync(dir, 'utf-8');
+        const versionRegex = /Version:\s*([\d]+\.[\d]+\.[\d]+(?:-[a-z]+\.[\d]+)?)/i;
+        const match = content.match(versionRegex);
+        if (match) {
+          const oldVersion = match[1];
+          const newVersion = incrementVersionString(oldVersion, type);
+          content = content.replace(versionRegex, `Version: ${newVersion}`);
+          fs.writeFileSync(dir, content, 'utf-8');
+          terminal(`#${dir}: ${oldVersion} -> ${newVersion}\n`);
+        }
+      } catch (err) {
+        console.error(`Error incrementing version in ${dir}: ${err.message}`);
+      }
+    }
+  }
+
+  /**
+   * Function to increment the current version.
+   * @param {string} version - current version
+   * @param {string} type - type of version to increase.
+   * @returns {string} version - updated version
+   */
+  function incrementVersionString(version, type) {
+    if (!semver.valid(version)) {
+      throw new Error(`Invalid version: ${version}`);
+    }
+    const [baseType, prerelease] = type.split(':');
+    if (type === 'beta' && semver.prerelease(version)) {
+      return semver.inc(version, 'prerelease', 'beta');
+    }
+    if (prerelease === 'beta') {
+      // Handle pre-release increment
+      if (semver.prerelease(version)) {
+        switch (baseType) {
+          case 'major':
+            return semver.inc(version, 'premajor', 'beta');
+          case 'minor':
+            return semver.inc(version, 'preminor', 'beta');
+          case 'patch':
+            return semver.inc(version, 'prepatch', 'beta');
+          default:
+            throw new Error(`Could not change prerelease version`);
+        }
+      }
+      // Convert stable version to pre-release
+      return semver.inc(`${semver.inc(version, baseType)}-beta.0`, 'prerelease', 'beta');
```

Comment:
@ampersarnie do you mean line 228 or 213 through to 228? 

Lines 213 to 226 convert the wording we use major, minor etc to the words which semver use premajor for a beta version. 

Line 228 allows you to convert a major release for example 1.0.0 back into beta

Reply:
Ok I got it. Thanks for the explanation.

In terms of line 229, it would be better to use the inbuilt functionality of the `semver.inc()`, you can define the beta version by assigning the value in the 4th parameter for the prerelease base version. See documentation here: https://www.npmjs.com/package/semver#prerelease-identifier-base

Reply:
I've added that in now thank you for that advice! http://bigbite.im/i/J1a7bi

Reply:
Thanks. I've left additional feedback on this in a separate review comment so will resolve this one.